tensorflow/compiler/jit/mark_for_compilation_pass.cc

```diff
@@ -923,20 +923,35 @@ MarkForCompilationPassImpl::ClusteringWillIntroduceInterDeviceDependency(
 }
 
 absl::optional<string> MarkForCompilationPassImpl::GetXlaScope(Node* node) {
-  // Look for an _XlaScope on both nodes.  If both nodes have a scope and the
-  // scopes do not match, do not cluster along this edge. This restriction is
-  // overridden if the global_jit_level_ is ON. If even one of the nodes lacks
-  // an _XlaScope attribute, then it is treated as a "bridge" and a cluster may
-  // be created along it.  We may want to restrict this behavior to require all
-  // nodes marked with _XlaCompile=true to also have a _XlaScope property set
-  // (and raise an error otherwise); but for now we don't do this.
-  if (global_jit_level_ != OptimizerOptions::OFF) {
-    return absl::nullopt;
-  }
+  // Look for either _XlaScope or _XlaAutoJitScope on both nodes to guide
+  // clustering.  If both nodes have a scope and the scopes do not match, do
+  // not cluster along this edge.  If even one of the nodes lacks a scope
+  // attribute, then it is treated as a "bridge" and a cluster may be created
+  // along it.
+  //
+  // The difference between _XlaScope and _XlaAutoJitScope is that _XlaScope is
+  // provided by users through jit_scope APIs, while _XlaAutoJitScope is
+  // automatically generated by the ClusterScopingPass when auto_jit is on.  As
+  // such, we respect _kXlaScope only when auto_jit is off, while respecting
```

Comment:
Either s/_kXlaScope/_XlaScope/ or s/_kXlaScope/kXlaScope/

Same for _kXlaAutoJitScope